elftools/dwarf/ranges.py

```diff
@@ -12,18 +12,42 @@
 from ..common.utils import struct_parse
 
 
-RangeEntry = namedtuple('RangeEntry', 'begin_offset end_offset')
-BaseAddressEntry = namedtuple('BaseAddressEntry', 'base_address')
+RangeEntry = namedtuple('RangeEntry', 'entry_offset entry_length begin_offset end_offset is_absolute')
+BaseAddressEntry = namedtuple('BaseAddressEntry', 'entry_offset base_address')
 
+def not_implemented(e):
+    raise NotImplementedError("Range list entry %s is not supported yet" % (e.entry_type,))
+
+# Maps parsed entry types to RangeEntry/BaseAddressEntry objects
+entry_translate = {
+    'DW_RLE_base_address' : lambda e: BaseAddressEntry(e.entry_offset, e.address),
+    'DW_RLE_offset_pair'  : lambda e: RangeEntry(e.entry_offset, e.entry_length, e.start_offset, e.end_offset, False),
+    'DW_RLE_start_end'    : lambda e: RangeEntry(e.entry_offset, e.entry_length, e.start_address, e.end_address, True),
+    'DW_RLE_start_length' : lambda e: RangeEntry(e.entry_offset, e.entry_length, e.start_address, e.start_address + e.length, True),
+    'DW_RLE_base_addressx': not_implemented,
+    'DW_RLE_startx_endx'  : not_implemented,
+    'DW_RLE_startx_length': not_implemented
+}
 
 class RangeLists(object):
     """ A single range list is a Python list consisting of RangeEntry or
         BaseAddressEntry objects.
+
+        version is used to distinguish DWARFv5 rnglists section from
+        the DWARF<=4 ranges section. Only the 4/5 distinction matters.
+
+        The dwarfinfo is needed for enumeration, because enumeration
+        requires scanning the DIEs, because ranges may overlap, even on DWARF<=4
     """
-    def __init__(self, stream, structs):
+    # Since dwarfinfo is not a required parameter, there is fallback to the
```

Comment:
I'm OK with making it mandatory and simplifying the code (no separate fallback logic). Passing an extra dwarfinfo shouldn't be an issue if someone finds this troublesome.

Reply:
The natural way to get to RangeLists is via DWARFInfo.range_lists() anyway.